posthog/hogql/printer.py

```diff
@@ -454,6 +461,18 @@ def visit_property_ref(self, ref: ast.PropertyRef):
 
         return property_sql
 
+    def visit_sample_expr(self, ref: ast.SampleExpr):
+        sample_value = self.visit_ratio_expr(ref.sample_value)
+        offset_clause = ""
+        if ref.offset_value:
+            offset_value = self.visit_ratio_expr(ref.offset_value)
+            offset_clause = f" OFFSET {offset_value}"
+
+        return f"SAMPLE {sample_value}{offset_clause}"
+
+    def visit_ratio_expr(self, ref: ast.RatioExpr):
```

Comment:
These are not really "ref"-s but just nodes. All the `visit` functions below are for "ref"-s.

```suggestion
    def visit_sample_expr(self, node: ast.SampleExpr):
        sample_value = self.visit_ratio_expr(ref.sample_value)
        offset_clause = ""
        if ref.offset_value:
            offset_value = self.visit_ratio_expr(ref.offset_value)
            offset_clause = f" OFFSET {offset_value}"

        return f"SAMPLE {sample_value}{offset_clause}"

    def visit_ratio_expr(self, node: ast.RatioExpr):
```